### DISCUSSION TITLE
Drop support for old CMake versions

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#25379] [Ideas]
__Why?__
- vcpkg internally uses a rather new CMake version
- if users want to use CMake for projects and integrate vcpkg they literally have to use the same version to get exactly the same behavior. (e.g. if `Find<Package>.cmake` modules are updated. Example: CURL #6386. Supporting older CMake versions here would require conditional changes to the library name which on the other hand breaks newer CMake behavior by not finding debug libraries)
- so compatibility with older/others versions of CMake never really existed for external projects. 
- CI tests are probably only performed with the internally used version of CMake.

So in the end VCPKG can only promise that everything works correctly with the CMake version itself uses to build packages. So vcpkg should at least issue a warning in the toolchain file if the version of CMake used within vcpkg and the version used externally disagree.

**Comment by cbezault** [comment]
The right thing to do is use cmake_minimum_required in order to have the correct cmake policies in place. While this is not 100% foolproof as far as backwards compatibility goes if it doesn't work, it's likely a bug and should be reported upstream. Therefore it's not a problem if we use a newer version of cmake, it's a problem if a port/library doesn't declare a minimum version.

**Comment by cbezault** [comment]
Please excuse my rushed comment, I didn't read the issue closely enough. I see that the PR you're referring to is concerned with the FindPackage.cmake files shipped with cmake. I'll have to think about that.

**Comment by BillyONeal** [comment]
@JackBoosY Why is this "requires: discussion"?

**Comment by JackBoosY** [comment]
@BillyONeal In the tool related code change history in vcpkg, we have never strictly dealt with the version of the tool.
Obviously, for some tools, the version update changes the API and functions to a greater extent, which leads to such problems.

**Comment by PhoebeHui** [comment]
@Neumann-A, I think we have improvements now, does it match your original request?

https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/cmake-guidelines.md#cmake-versions-to-require
https://github.com/microsoft/vcpkg/blob/master/scripts/ports.cmake

**Comment by Neumann-A** [comment]
@PhoebeHui : This is about the externally used cmake version not about the internally used one.

**Comment by JackBoosY** [comment]
Can we use cmake policy to avoid this?
